app/services/PaymentAPIService.scala

```diff
@@ -64,25 +64,56 @@ class PaymentAPIService(wsClient: WSClient, paymentAPIUrl: String) {
       .execute()
   }
 
-  def isDuplicatePaymentResponse(response: WSResponse): Boolean = {
+  def decodePaymentResponse(response: WSResponse): Option[PaymentApiError] = {
     decode[PaymentApiError](response.body).fold(
       failure => {
         val failureMessage = SafeLogger.LogMessage(
```

Comment:
I know you didn't add it in this PR, but I would use scrub too - we don't really want people building a LogMessage directly.

Reply:
Do we have to scrub if we e.g. put/have a retention policy of 2 weeks on our log messages? In this instance, doesn't scrubbing limit the usefulness of the log message?

Reply:
good idea!

Reply:
Its for sentry I think - as far as I can tell scrub redacts the logs that go to sentry but keeps the data in our AWS logs

Reply:
nice!

Reply:
Yeah, that's right @michaelwmcnamara 👍